### DISCUSSION TITLE
Panels don't contain same tabs across different FF windows?

### DISCUSSION THREAD

**Discussion by davidleejy** [mbnuqw/sidebery#1285] [Q&A]
Hi fellow Sidebery users,

Sidebery version: 5.0.0
Firefox version: 117.0.1

I'm trying to use panels for the first time. 

Previously I relied on separate FF windows to separate my tabs for different tasks. For example, I might have one FF window for my studies, another window for work-related tabs, and one other for personal finance tabs:

<img width="318" alt="Screenshot 2023-09-27 at 2 04 10 PM" src="https://github.com/mbnuqw/sidebery/assets/1690072/89e741c4-2e19-4c6e-aa69-4a30d074becc">

Wanting to have fewer FF windows opened, I decided to make the transition to using Sidebery panels instead to separate tabs concerning different tasks.

I created three panels named "studies", "work", and "finance" and moved my tabs into their respective panels. However, to my confusion, these panels would be empty in newly opened FF windows.

Hold up, I said to myself, shouldn't all the identically named panels contain the same tabs across all FF windows especially since these panels are visible in all FF windows? Is my intuition about how panels should be used somehow wrong?

My intuition:

```Typst
FF window 1:
  Panel "studies": {tab1, tab2, tab3, tab4}
  Panel "work": {tab5, tab6, tab7}
  Panel "personal finance": {tab8, tab9, tab10, tab11}

FF window 2 (newly opened):
  Panel "studies": {tab1, tab2, tab3, tab4}
  Panel "work": {tab5, tab6, tab7}
  Panel "personal finance": {tab8, tab9, tab10, tab11}
```

My reality:

```Typst
FF window 1:
  Panel "studies": {tab1, tab2, tab3, tab4}
  Panel "work": {tab5, tab6, tab7}
  Panel "personal finance": {tab8, tab9, tab10, tab11}

FF window 2 (newly opened):
  Panel "studies": {}
  Panel "work": {}
  Panel "personal finance": {}
```

**Comment by llc0930** [comment]
In fact, I personally think it is more like a label.
In addition, if you right-click a tab on the panel of window 1 and send it to another window, it will be sent to the same panel.

**Comment by davidleejy** [comment]
Ah ok I think I'm getting the idea.. If a panel is like a label, can it also function similarly to a tag? Like, can users conveniently assign a single tab to multiple panels, similar to how in general a single item can possess multiple tags?

**Comment by llc0930** [comment]
Unfortunately, it doesn't seem to meet your requirements. I guess I should correct my statement. It seems to be a unique label.

**Comment by krafd** [comment]
I know this is a year old, but I just realized the same thing.
Can anyone please confirm if this is still the case, or, perhaps something has been added to the config?

If I have a secondary FF window open, and add a tab to a particular panel, do I basically lose it if I forget to send it over to main?  Can be very frustrating if one doesn't notice at the time.

Thanks

Note: Using FF 131.0.2 - archlinux

**Reply by llc0930** [reply]
Not sure, I rarely open two windows, so I don't pay close attention to it.
If you haven't cleared your history then you may be able to find it, otherwise good luck to you my friend.

**Reply by seppulcro** [reply]
> If I have a secondary FF window open, and add a tab to a particular panel, do I basically lose it if I forget to send it over to main? Can be very frustrating if one doesn't notice at the time.

I've just tested it on MacOS Sonoma 14.6.1

1. With 2 firefox windows, opened a tab on my secondary monitor window 2 on panel 2.
2. Command+Q to exit firefox.
3. Re-open Firefox, launches 2 windows. Each with it's own sidebery tabs and state's restored.

Isn't this the default behaviour without sidebery?
If you close window 2 and not restore it, then close window 1 thus quitting firefox, window 1 was the only window open when the save state before quit happens.

**Reply by llc0930** [reply]
Yes, in my understanding, Sidebery has never managed the storage state, but only records the tree level. Sidebery's snapshot function is independent of Firefox's state management. Please remind me if my understanding is wrong.

**Reply by krafd** [reply]
thank you all for the replies.
I will just be more aware/careful when opening additional windows :)

**Reply by ChilledMoth** [reply]
> thank you all for the replies. I will just be more aware/careful when opening additional windows :)

A new feature that moves all tabs from the active window to another one at once (while respecting their panel associations) might be useful. Ideally with the option to do so automatically on the closure of a window as long as there is at least one other window open. Consolidation of all windows into a single one on the push of a button would be only one step away.

**Reply by krafd** [reply]
> > thank you all for the replies. I will just be more aware/careful when opening additional windows :)
> 
> A new feature that moves all tabs from the active window to another one at once (while respecting their panel associations) might be useful. Ideally with the option to do so automatically on the closure of a window as long as there is at least one other window open. Consolidation of all windows into a single one on the push of a button would be only one step away.

You know what... I haven't done any proper coding in years.. but I think you just gave me an excuse to do a little side-project :)

I'll follow up on this thread if/when I get something implemented.

**Reply by llc0930** [reply]
Moving to a new window will move it into the first panel, and moving to another existing window will move it into the corresponding panel, so it should be very close, just one step away.

**Comment by rublev** [comment]
Any updates on this?

**Reply by llc0930** [reply]
Because Firefox's browsing state backup mechanism isn't perfect, a power outage while writing a backup often results in corrupted backup files and the loss of numerous tabs. Therefore, I enable scheduled automatic snapshot exports. These snapshots record all tabs in all windows, preventing tab loss. This also prevents accidental window closings, right?

**Reply by rublev** [reply]
Got it, I did the same. Thanks.

**Reply by llc0930** [reply]
It would be great if this helps you too, I've lost tabs so many times in the past because I was too lazy to add UPS...

**Reply by rublev** [reply]
> It would be great if this helps you too, I've lost tabs so many times in the past because I was too lazy to add UPS...

The snapshot feature absolutely rocks. Thanks for the suggestion!
